### PR TITLE
feat: WrappedPreBlocker

### PR DESCRIPTION
**Pull request by technicallyty**
previously, simapp would not work with x/upgrade handlers. this is because upgrades rely on `PreBlocker` but this was being overwritten with the oracle PreBlocker.

fixed by creating a new method on oracle's PreBlockHandler: "WrappedPreBlocker" which takes a `module.Manager`, and call the PreBlock function to trigger upgrades. h/t @aljo242 for showing me the way

i added a comment to the existing PreBlocker that notes it is deprecated and to use WrappedPreBlocker instead.

### CODE REVIEW

File: tests/simapp/app.go [hunk 1]
```diff
@@ -530,6 +531,28 @@ func (app *SimApp) RegisterAPIRoutes(apiSvr *api.Server, apiConfig config.APICon
 	}
 }
 
+// wrapAndSetPreBlocker sets the application's PreBlocker while still calling the originals.
+// This is needed to set custom PreBlock logic, but still allow things like x/upgrade to function.
+func (app *SimApp) wrapAndSetPreBlocker(pb sdk.PreBlocker) {
```
Comment (g3ndo9kar9) [inline tests/simapp/app.go]: Let's export this fn publicly? This is an operation all integrators have to roll on their own currently

**Review by g3ndo9kar9 — APPROVED** [review]
lgtm

File: tests/simapp/app.go (hunk 1 above)
Comment (technicallyty) [reply tests/simapp/app.go]: i could make it so `oraclePreBlockHandler.PreBlocker()` takes a `*module.Manager` argument, so this does it automatically? does that sound good?

Comment (aljo242) [reply tests/simapp/app.go]: Yeah, that makes sense to me

Comment (technicallyty) [reply tests/simapp/app.go]: updated PR to reflect the change, let me know if all is good @nivasan1 @aljo242

**Review by davidterpay — APPROVED** [review]

File: abci/preblock/oracle/preblock_test.go
Comment (technicallyty) [inline abci/preblock/oracle/preblock_test.go:1]: i just switched all the tests over to the non-deprecated version of the method, and added a case to ensure `PreBlock` from `ModuleManager` is actually called. lmk if this is ideal or not!

**Review by aljo242 — APPROVED** [review]

**Pull request merged**